walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -172,32 +222,68 @@ public static void addAvailableTime(int start, int prevEnd, int duration,
       return new ArrayList<TimeRange>(Arrays.asList(wholeDay));
     }
     
-    // Filter events input
     Event[] eventsArray = new Event[events.size()];
     eventsArray = events.toArray(eventsArray);
 
-    // remove all events that do not have the attendees from the meeting request 
-    Predicate<Event> isIntersection = new IsIntersection(request.getAttendees());
-    eventsArray = includeEventIf(eventsArray, isIntersection);
-
-    // sort the events that are remaining
+    IncludeIf<Event> includeIf = new IncludeIf<Event>();
     MergeSort<Event> merge = new MergeSort<Event>();
-    merge.sort(eventsArray, new SortEventsByTime());
 
-    // Get the times of the event and remove events with the same times
-    //    (but keep the longest duration) and the events that are contained in 
-    //    another event
-    ArrayList<int[]> times = new ArrayList<int[]>();
+    // filter and sort the events that mandatory attendees are attending
+    Predicate<Event> isMandatoryIntersection = new IsIntersection
+        (request.getAttendees());
+    
+    // Filter
+    ArrayList<Event> filteredMandatoryEvents = includeIf.includeIf
+        (eventsArray, isMandatoryIntersection);
+
+    // Sort
+    merge.sort(filteredMandatoryEvents, new SortEventsByTime());
+
+    // Turn events into int[]
+    ArrayList<int[]> mandatoryTimes = new ArrayList<int[]>();
     try {
-      times = eventToTime(eventsArray);  
+      mandatoryTimes = eventToFilteredTime(filteredMandatoryEvents);  
     } catch (Exception e) {
       throw e;
     }
+    
+    // filter and sort the events that optional attendees are attending  
+    Predicate<Event> isOptionalIntersection = new IsIntersection
+        (request.getOptionalAttendees());   
 
-    // Compare filtered events input to meeting request
-    //    Find the time available for this meeting
-    Collection<TimeRange> availableTimes = timeAvailable(times, 
-        durationMeetingMinutes);
-    return availableTimes;
+    // Filter
+    ArrayList<Event> filteredOptionalEvents = includeIf.includeIf
+        (eventsArray, isOptionalIntersection);
+
+    // Sort
+    merge.sort(filteredOptionalEvents, new SortEventsByTime()); 
+     
+    // Turn events into int[]
+    ArrayList<int[]> optionalTimes = new ArrayList<int[]>();
+    try {
+      optionalTimes = eventToFilteredTime(filteredOptionalEvents);  
+    } catch (Exception e) {
+      throw e;
+    }    
```

Comment:
This try/catch statement is effective a no-op. If you're not doing anything with the exception, you don't have to catch it.

```suggestion
    optionalTimes = eventToFilteredTime(filteredOptionalEvents);  
```